operator/controllers/keptnevaluation/providers/dynatrace.go

```diff
@@ -0,0 +1,116 @@
+package providers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	klcv1alpha1 "github.com/keptn/lifecycle-toolkit/operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KeptnDynatraceProvider struct {
+	Log       logr.Logger
+	client    http.Client
+	k8sClient client.Client
+}
+
+type DynatraceResponse struct {
+	TotalCount int               `json:"totalCount"`
+	Resolution string            `json:"resolution"`
+	Result     []DynatraceResult `json:"result"`
+}
+
+type DynatraceResult struct {
+	MetricId string          `json:"metricId"`
+	Data     []DynatraceData `json:"data"`
+}
+
+type DynatraceData struct {
+	Timestamps []int64    `json:"timestamps"`
+	Values     []*float64 `json:"values"`
+}
+
+func (d *KeptnDynatraceProvider) EvaluateQuery(ctx context.Context, objective klcv1alpha1.Objective, provider klcv1alpha1.KeptnEvaluationProvider) (string, error) {
+	qURL := provider.Spec.TargetServer + "/api/v2/metrics/query?metricSelector=" + objective.Query
+
+	d.Log.Info("Running query: " + qURL)
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, "GET", qURL, nil)
+	if err != nil {
+		d.Log.Error(err, "Error while creating request")
+		return "", err
+	}
+
+	token, err := d.getDTApiToken(ctx, provider)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Authorization", "Api-Token "+token)
+	res, err := d.client.Do(req)
+	if err != nil {
+		d.Log.Error(err, "Error while creating request")
+		return "", err
+	}
+	defer func() {
+		err := res.Body.Close()
+		if err != nil {
+			d.Log.Error(err, "Could not close request body")
+		}
+	}()
+
+	// we fail later while unmarshalling
```

Comment:
```suggestion
```

Reply:
I'll rewrite making explicit that I am ignoring the error for a reason